autoarray/plot/mat_wrap/wrap/wrap_2d.py

```diff
@@ -0,0 +1,568 @@
+        regions : list of tuples
+            Indices of vertices in each revised Voronoi regions.
+        vertices : list of tuples
+            GridIrregularGrouped for revised Voronoi vertices. Same as coordinates
+            of input vertices, with 'points at infinity' appended to the
+            end.
+        """
+
+        if voronoi.points.shape[1] != 2:
+            raise ValueError("Requires 2D input")
+
+        new_regions = []
+        new_vertices = voronoi.vertices.tolist()
+
+        center = voronoi.points.mean(axis=0)
+        if radius is None:
+            radius = voronoi.points.ptp().max() * 2
+
+        # Construct a map containing all ridges for a given point
+        all_ridges = {}
+        for (p1, p2), (v1, v2) in zip(voronoi.ridge_points, voronoi.ridge_vertices):
+            all_ridges.setdefault(p1, []).append((p2, v1, v2))
+            all_ridges.setdefault(p2, []).append((p1, v1, v2))
+
+        # Reconstruct infinite regions
+        for p1, region in enumerate(voronoi.point_region):
+            vertices = voronoi.regions[region]
+
+            if all(v >= 0 for v in vertices):
+                # finite region
+                new_regions.append(vertices)
+                continue
+
+            # reconstruct a non-finite region
+            ridges = all_ridges[p1]
+            new_region = [v for v in vertices if v >= 0]
+
+            for p2, v1, v2 in ridges:
+                if v2 < 0:
+                    v1, v2 = v2, v1
+                if v1 >= 0:
+                    # finite ridge: already in the region
+                    continue
+
+                # Compute the missing endpoint of an infinite ridge
+
+                t = voronoi.points[p2] - voronoi.points[p1]  # tangent
+                t /= np.linalg.norm(t)
+                n = np.array([-t[1], t[0]])  # hyper
+
+                midpoint = voronoi.points[[p1, p2]].mean(axis=0)
+                direction = np.sign(np.dot(midpoint - center, n)) * n
+                far_point = voronoi.vertices[v2] + direction * radius
+
+                new_region.append(len(new_vertices))
+                new_vertices.append(far_point.tolist())
+
+            # sort region counterclockwise
+            vs = np.asarray([new_vertices[v] for v in new_region])
+            c = vs.mean(axis=0)
+            angles = np.arctan2(vs[:, 1] - c[1], vs[:, 0] - c[0])
+            new_region = np.array(new_region)[np.argsort(angles)]
+
+            # finish
+            new_regions.append(new_region.tolist())
+
+        return new_regions, np.asarray(new_vertices)
+
+
+class OriginScatter(GridScatter):
+    """
+    Plots the (y,x) coordinates of the origin of a data structure (e.g. as a black cross).
+
+    See `mat_structure.Scatter` for a description of how matplotlib is wrapped to make this plot.
+    """
+
+    pass
+
+
+class MaskScatter(GridScatter):
+    """
+    Plots a mask over an image, using the `Mask2d` object's (y,x) `edge_grid_sub_1` property.

```

Comment:
Why is it necessary to define all these classes? Could you not just use GridScatter for all these usecases?

Reply:
The class name is used to load default values from configs, so that is why.